src/bayesian_linear_regression.jl

```diff
@@ -73,3 +73,25 @@ function __compute_inference_quantities(fx::FiniteBLR, y::AbstractVector{<:Real}
 
     return Uw, Bt, δy, logpdf_δy, Λεy
 end
+
+# Random function sample generation
+# Following the Random API: https://docs.julialang.org/en/v1/stdlib/Random/#An-optimized-sampler-with-pre-computed-data
+struct BLRFunctionSample{Tw<:AbstractVector}
+    w::Tw
+end
+
+(s::BLRFunctionSample)(X::AbstractMatrix{<:Real}) = X's.w
+(s::BLRFunctionSample)(X::ColVecs) = X.X's.w
+(s::BLRFunctionSample)(X::RowVecs) = X.X * s.w
```

Comment:
Since the package assumes that `X <: AbstractMatrix` - are the ColVecs/RowVecs methods needed?

Reply:
Actually - might be worth it to make the entire package work with either ColVecs/RowVecs?

Reply:
I don't actually recall _why_ I'm using matrices all over the place in this package.... definitely something we should change for the sake of consistency.

Reply:
#25